### PR TITLE
Fractal encounters & general cleanup

### PR DESCRIPTION
**Pull request by Linkaaaaa**
# General
- Added Flux Bomb target buff
- Added Toxic Sickness mechanic
- Cleaned up ArcDPSEnums for some encounters
- Fixed Toxic Sickness decoration not working with the new effect
- Added Flux Bomb on player decoration
- Cleaned up lifespans and removed a bunch of no longer necessary (int) casts
- Added base.ComputeEnvironmentCombatReplayDecorations(log) to some missing Environment overrides
- Moved effect decorations to the environment combat replay for the mechanics that made sense doing so
- Moved Caustic Barrage and Solar Bolt to Fractal Logic

# MAMA
- Corrected Blastwave radius size
- Corrected Leap radius size
- Split the Knights jump and fall attacks into 2 different decorations
- Split Nightmare Miasma into two decorations

# Siax
- Adjusted duration of the orb AoEs during some mechanics

# Ensolyss
- Added Rampage decoration
- Added AoEs during the Nightmare Altar phases
- Adjusted duration of the orb AoEs during some mechanics

# Arkk
- Compacted Horizon Strike decoration

# API
- Converted ComputeEndCastTimeByStun to generic ComputeEndCastTimeByBuffApplication
- Cleaned up Fractal encounters that were using the above to the new API
- Added a couple new colors
- Allow ComputeLifespanWithSecondaryEffectAndPosition to accept a different minimum distance than default

### CODE REVIEW

File: GW2EIEvtcParser/EncounterLogic/Fractals/FractalLogic.cs [hunk 1]
```diff
@@ -146,26 +147,41 @@ protected static long GetFightOffsetByFirstInvulFilter(FightData fightData, Agen
         internal override void ComputePlayerCombatReplayActors(AbstractPlayer p, ParsedEvtcLog log, CombatReplay replay)
         {
             base.ComputePlayerCombatReplayActors(p, log, replay);
-            if (log.CombatData.TryGetEffectEventsByGUID(EffectGUIDs.ToxicSicknessPuke1, out IReadOnlyList<EffectEvent> sicknessEffects))
+
+            // Toxic Sickness - Old Indicator
```
Comment (EliphasNUIT) [inline GW2EIEvtcParser/EncounterLogic/Fractals/FractalLogic.cs:151]: var toxicSicknessGUIDs = new List<String>() { EffectGUIDs.ToxicSicknessOldIndicator, EffectGUIDs.ToxicSicknessNewIndicator};
then iterate over that list, the content of the 2 ifs is identical.

File: GW2EIEvtcParser/EncounterLogic/Fractals/Nightmare/Siax.cs [hunk 2]
```diff
@@ -269,5 +213,80 @@ internal override void ComputePlayerCombatReplayActors(AbstractPlayer p, ParsedE
...
+                foreach (EffectEvent effect in spawnEffects)
+                {
+                    var circle = new CircleDecoration(360, (effect.Time, effect.Time + duration), Colors.Orange, 0.2, new PositionConnector(effect.Position));
+                    EnvironmentDecorations.Add(circle);
+                    EnvironmentDecorations.Add(circle.Copy().UsingGrowingEnd(effect.Time + duration));
+                }
+            }
+
+            // Caustic Barrage
+            if (log.CombatData.TryGetEffectEventsByGUID(EffectGUIDs.CausticBarrageIndicator, out IReadOnlyList<EffectEvent> barrageEffects))
```
Comment (EliphasNUIT) [inline GW2EIEvtcParser/EncounterLogic/Fractals/Nightmare/Siax.cs:255]: Barring a few numerical values, this is exactly the same code as in Ensolyss, can be refactored.

File: GW2EIEvtcParser/EncounterLogic/Fractals/ShatteredObservatory/Skorvald.cs [hunk 3]
```diff
@@ -590,15 +533,60 @@ internal override void ComputeEnvironmentCombatReplayDecorations(ParsedEvtcLog l
...
+                    (long start, long end) lifespan = effect.ComputeLifespan(log, 1000);
+                    EnvironmentDecorations.Add(new CircleDecoration(130, lifespan, Colors.Orange, 0.2, new PositionConnector(effect.Position)));
+                }
+            }
+
+            // Solar Bolt - Indicator
+            if (log.CombatData.TryGetEffectEventsByGUID(EffectGUIDs.SolarBoltIndicators, out IReadOnlyList<EffectEvent> solarBoltIndicators))
             {
-                foreach (EffectEvent mistBombEffect in mistBombEffects)
+                foreach (EffectEvent effect in solarBoltIndicators)
```
Comment (EliphasNUIT) [inline GW2EIEvtcParser/EncounterLogic/Fractals/ShatteredObservatory/Skorvald.cs:548]: Same code as in Ensolyss and Siax.

Comment (Linkaaaaa) [reply GW2EIEvtcParser/EncounterLogic/Fractals/ShatteredObservatory/Skorvald.cs:548]: should i call it "AddCausticBarrageOrSolarBoltDecoration" or got something better?

Comment (EliphasNUIT) [reply GW2EIEvtcParser/EncounterLogic/Fractals/ShatteredObservatory/Skorvald.cs:548]: AddDistanceCorrectedOrbDecorations?

Comment (EliphasNUIT) [reply GW2EIEvtcParser/EncounterLogic/Fractals/ShatteredObservatory/Skorvald.cs:548]: (ParsedEvtcLog log, int targetID, double distanceThreshold, double onDistanceSuccessDuration, double onDistanceFailDuration)

Comment (Linkaaaaa) [reply GW2EIEvtcParser/EncounterLogic/Fractals/ShatteredObservatory/Skorvald.cs:548]: Do you want it located inside fractal logic or combat replay?

Comment (EliphasNUIT) [reply GW2EIEvtcParser/EncounterLogic/Fractals/ShatteredObservatory/Skorvald.cs:548]: FractalLogic, at least for now

**Comment by EliphasNUIT** [comment]
Thanks!

**Pull request merged**